legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/NontemporalDeltaPlanner.java

```diff
@@ -132,13 +177,22 @@ private Merge getMergeOperation(Condition pkMatchCondition, Condition digestDoes
             stagingDataset = Selection.builder().source(stagingDataset()).condition(dataSplitInRangeCondition).addAllFields(fieldsToSelect).alias(stagingDataset().datasetReference().alias()).build();
         }
 
+        if (deleteIndicatorIsNotSetCondition.isPresent())
+        {
+            digestCondition = Or.builder().addConditions(digestDoesNotMatchCondition, deleteIndicatorIsNotSetCondition.get()).build();
```

Comment:
Should it be OR or AND? if using OR, we will try to update the record even if it is identical (digest same)

Reply:
should be AND got this condition from unitemp where the timestamp has to be updated but here it should be AND